main/can.c

```diff
@@ -61,33 +63,36 @@ void CAN_Close()
 
 // For recieving a message. Blocks for up to {timeout} ms. 0 for failure, 1 for success.
 uint8_t CAN_ReceiveMsg(CanMessage_t * msg, int timeout){
-    can_message_t _msg;
-    printf("%s:%d dequeueing message\n", __FILE__, __LINE__);
-    esp_err_t err = can_receive(&_msg, portTICK_PERIOD_MS * timeout);
-    printf("%s:%d dequeued message sucessfully\n", __FILE__, __LINE__);
+    twai_message_t _msg;
+    //printf("%s:%d dequeueing message\n", __FILE__, __LINE__);
+    esp_err_t err = twai_receive(&_msg, portTICK_PERIOD_MS * timeout);
+    //printf("%s:%d dequeued message sucessfully\n", __FILE__, __LINE__);
     if(err == ESP_OK){
         msg->id = _msg.identifier;
         msg->len = _msg.data_length_code;
         memmove(msg->payload, _msg.data, 8);
-        printf("%s:%d successfully got the message\n", __FILE__, __LINE__);
+        //printf("%s:%d successfully got the message\n", __FILE__, __LINE__);
         return 1; //sucessful message RX
     }
-    printf("%s:%d FAILED: fetching message\n", __FILE__, __LINE__);
+    printf("%s:%d FAILED: %x fetching message\n", __FILE__, __LINE__,err);
     return 0; //something terrible happened
 }
 
 // TODO: understand error codes for this one - 255 appears to be sucess and 0 failure.
 uint8_t CAN_SendMsg(CanMessage_t * msg){
-    can_message_t _msg;
+    twai_message_t _msg;
     _msg.identifier = msg->id;
     _msg.data_length_code = msg->len;
+    _msg.flags = 0;

```

Comment:
Flags is deprecated, also worked with `_msg.rtr = 0;` and `_msg.extd = 0;`.
I think only clearing the memory is necessary here, to remove the garbage, nothing special.